lib/ProductOpener/Products.pm

```diff
@@ -632,6 +651,49 @@ sub compute_sort_key($) {
 	# otherwise sort by last_modified_t (e.g.  1571384133)
 
 	my $sortkey = $product_ref->{last_modified_t};
+	
+	my $popularity_key = 0;
+	
+	# Use the popularity tags
+	if (defined $product_ref->{popularity_tags}) {
+		my %years = ();
+		my $latest_year;
+		foreach my $tag (@{$product_ref->{popularity_tags}}) {
+			# one product could have:
+			# "top-50000-scans-2019",
+			# "top-100000-scans-2019",
+			# "top-100000-scans-2020",
+			if ($tag =~ /^top-(\d+)-scans-20(\d\d)$/) {
+				my $top = $1;
+				my $year = $2;
+				# Save the smaller top for each year
+				if ((not defined $years{$year}) or ($years{$year} > $top)) {
+					$years{$year} = $top;
+				}
+				if ((not defined $latest_year) or ($year > $latest_year)) {
+					$latest_year = $year;
+				}
+			}
+		}
+		# Keep only the latest year, and make the latest year count more than previous years
+		if (defined $latest_year) {
+			$popularity_key += $latest_year * 1000000 * 1000 - $years{$latest_year} * 1000;
+		}
+	}
+	
+	# unique_scans_n : number of unique scans for the last year processed by scanbot.pl
+	if (defined $product_ref->{unique_scans_n}) {
+		$popularity_key += $product_ref->{unique_scans_n};
+	}
+	
+	# give a small boost to products for which we have recent images
+	if (defined $product_ref->{last_image_t}) {
+
+		my $age = int((time() - $product_ref->{last_image_t}) / (86400 * 30));	# in months
+		if ($age < 12) {
+			$popularity_key += 12 - $age;
+		}
+	}
 
 	if ((not defined $product_ref->{obsolete}) or (not $product_ref->{obsolete})) {
 		$sortkey += 200000000000;
```

Comment:
I don't really like these really big numbers, expecially as mongo index 😕. Could subtracting be an idea? Does mongo deals with negative indexes?

Reply:
I think negative numbers should work. I do not know if there's an issue with big numbers though.

Reply:
I changed the keys to make obsolete products negative. We still have big numbers, but I don't really think that's an issue for mongo, it seems to work quite well. We can revisit the keys later, we will probably have to change them anyway.